### ISSUE TITLE
About loss coefficient settings in pretraining

### ISSUE THREAD

**Issue by xxh-1998** [Arcee-LYK/Multi-to-Single#4]
Hi,

I noticed during pretraining that mi_loss is multiplied by 0.3 and recon_loss is multiplied by 0.1. This seems to make their magnitudes about two orders of magnitude smaller than contra_loss and cpc_loss.

I have a couple of questions regarding this:

Were these coefficients (0.3 for mi_loss, 0.1 for recon_loss) determined through hyperparameter grid search, or should I consider adjusting them to bring all losses to a similar order of magnitude?
Additionally, I observed that mi_loss can become negative at the beginning of training. Is this expected behavior?

**Comment by Arcee-LYK** [comment]
The recon loss of 0.1 is recommended in the original VQ-VAE paper. The coefficient of mi loss should have been referenced, but I don’t remember it clearly. I haven’t paid attention to the specific loss reduction.
